File: boot-server/src/main/java/org/paratranz/bot/mirai/command/AuditCommandProcessor.java
package org.paratranz.bot.mirai.command;

import cn.hutool.core.collection.CollUtil;
import cn.hutool.core.util.StrUtil;
import org.paratranz.bot.mirai.command.annotation.CommandProcessor;
import org.paratranz.bot.mirai.command.core.AbstractCommandProcessor;
import org.paratranz.bot.para.entity.Page;
import org.paratranz.bot.para.entity.PageResult;
import org.paratranz.bot.para.entity.data.Application;
import org.paratranz.bot.para.entity.data.Audit;
import org.paratranz.bot.para.server.ParaApiService;
import org.paratranz.bot.properties.ExternalProperties;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import net.mamoe.mirai.event.events.GroupMessageEvent;
import net.mamoe.mirai.message.data.MessageChainBuilder;

import java.util.List;


/**
 * 群审核指令处理器
 * @author Ankol
 */
@Slf4j
@RequiredArgsConstructor
@CommandProcessor("/群审核")
public class AuditCommandProcessor extends AbstractCommandProcessor<GroupMessageEvent> {

    private final ParaApiService paraApiService;
    private final ExternalProperties properties;

    @Override
    protected void onNoArgsEvent(GroupMessageEvent event) {
        PageResult<Application> pageResult = paraApiService.listApplications(new Page(), properties.getProjectId(), 0);
        if (CollUtil.isNotEmpty(pageResult.getResults())) {
            List<Application> results = pageResult.getResults();
            //有多个待审核人的情况
            if (results.size() > 1) {
//                multiprocessing(event, results);
                log.error("Error. application count > 1, {}", results);
            }
            //针对只有一个需要审核的人的情况.
            else {
                Application application = results.get(0);
                //状态0为待审核.这里做一个申请状态验证.
                if (application.getStatus() != 0) {
                    return;
                }
                //获取其测试内容
                List<Audit> testContent = paraApiService.getTestContent(application.getId(), properties.getProjectId());
                this.sendMessage(event, testContent);
            }
        } else {
            event.getGroup().sendMessage("抱歉，没找到需要审核人员呢~");
        }
    }

    @Override
    protected void onArgsEvent(GroupMessageEvent event, String[] args) {
        String applicationId = args[0];
        if (StrUtil.isNotEmpty(applicationId)) {
            try {
                List<Audit> content = paraApiService.getTestContent(Integer.parseInt(applicationId), properties.getProjectId());
                this.sendMessage(event, content);
            } catch (NumberFormatException e) {
                event.getGroup().sendMessage("无法解析参数，确保查询参数为正确的数字用户id。例：/群审核 [用户id]");
            }
        } else {
            event.getGroup().sendMessage("查询的用户id不可为空！例：/群审核 [用户id]");
        }
    }

    private void sendMessage(GroupMessageEvent event, List<Audit> testContent) {
        if (testContent.isEmpty()) {
            event.getGroup().sendMessage("未找到相关测试内容呢~");
        } else {
            for (Audit audit : testContent) {
                MessageChainBuilder builder = new MessageChainBuilder()
                        .append(audit.getOriginal()).append("\n\n");
                //旧词条翻译者用户信息可能为空
                Audit.OriginDTO.UserDTO user = audit.getOrigin().getUser();
                if (user != null) {
                    builder.append(user.getNickname() != null ? user.getNickname() : user.getUsername()).append("：").append("\n");
                }
                //继续拼接
                builder.append(audit.getOrigin().getTranslation()).append("\n\n")
                        .append("申请者翻译：").append("\n")
                        .append(audit.getTranslation());
                //发送消息
                event.getGroup().sendMessage(builder.build());
            }
        }
    }
}

File: server/src/main/java/org/paratranz/bot/bot/command/AuditCommandProcessor.kt
package org.paratranz.bot.bot.command

import cn.hutool.core.util.StrUtil
import kotlinx.coroutines.runBlocking
import net.mamoe.mirai.event.events.GroupMessageEvent
import net.mamoe.mirai.message.data.MessageChainBuilder
import org.paratranz.bot.api.ParatranzApi
import org.paratranz.bot.api.constant.ApplyStatus
import org.paratranz.bot.api.entity.Page
import org.paratranz.bot.api.entity.PageResult
import org.paratranz.bot.api.entity.data.Application
import org.paratranz.bot.api.entity.data.Audit
import org.paratranz.bot.bot.core.GroupMessageCommandProcessor
import org.paratranz.bot.properties.ExternalProperties
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

/**
 * /audit指令处理器
 *
 * @author Ankol
 */
@Component
class AuditCommandProcessor :
    GroupMessageCommandProcessor("群审核placeholder", arrayOf("/audit", "/群审核")) {

    private val log: Logger = LoggerFactory.getLogger(AuditCommandProcessor::class.java)

    @Autowired
    private lateinit var paraTranzApi: ParatranzApi

    @Autowired
    private lateinit var properties: ExternalProperties
    override fun onNoArgsEvent(event: GroupMessageEvent) {
        val pageResult: PageResult<Application> = paraTranzApi.apply.listApply(properties.projectId, Page.of(), ApplyStatus.UN_CONFIRM)
        if (!pageResult.results.isNullOrEmpty()) {
            val results = pageResult.results
            //有多个待审核人的情况
            if (results.size > 1) {
                log.error("Error. application count > 1, {}", results)
            } else {
                val application = results[0]
                //状态0为待审核.这里做一个申请状态验证.
                if (application.status != 0) {
                    return
                }
                //获取其测试内容
                val testContent = paraTranzApi.apply.getTestContent(
                    application.id, properties.projectId
                )
                sendMessage(event, testContent)
            }
        } else {
            runBlocking { event.group.sendMessage("抱歉，没找到需要审核人员呢~") }
        }
    }

    override fun onArgsEvent(event: GroupMessageEvent, args: Array<String>) {
        val applicationId = args[0]
        if (StrUtil.isNotEmpty(applicationId)) {
            try {
                val content = paraTranzApi.apply.getTestContent(applicationId.toInt(), properties.projectId)
                sendMessage(event, content)
            } catch (e: NumberFormatException) {
                runBlocking { event.group.sendMessage("无法解析参数，确保查询参数为正确的数字用户id。例：/群审核 [用户id]") }

            }
        } else {
            runBlocking { event.group.sendMessage("查询的用户id不可为空！例：/群审核 [用户id]") }
        }
    }

    private fun sendMessage(event: GroupMessageEvent, testContent: List<Audit>) {
        if (testContent.isEmpty()) {
            runBlocking { event.group.sendMessage("未找到相关测试内容呢~") }

        } else {
            for (audit in testContent) {
                val builder = MessageChainBuilder()
                    .append(audit.original).append("\n\n")
                //旧词条翻译者用户信息可能为空
                val user = audit.origin.user
                if (user != null) {
                    builder.append(if (user.nickname != null) user.nickname else user.username).append("：").append("\n")
                }
                //继续拼接
                builder.append(audit.origin.translation).append("\n\n")
                    .append("申请者翻译：").append("\n")
                    .append(audit.translation)
                //发送消息
                runBlocking { event.group.sendMessage(builder.build()) }
            }
        }
    }
}
